odl/tomo/data/mrc2014.py

```diff
@@ -0,0 +1,846 @@
+        according to [Che+2015]_ that the values are considered as
+        undetermined.
+    mrc_version : 2-tuple of int, optional
+        Version identifier for the MRC file, used for the ``'nversion'``
+        header entry.
+        Default: ``(2014, 0)``
+    text_labels : sequence of strings
+        Maximal 10 strings with 80 characters each, used for the
+        ``'nlabl'`` and ``'label'`` header entries.
+        Default: ``[]``
+
+    Returns
+    -------
+    header : `OrderedDict`
+        Header stored in an ordered dictionary, where each entry has the
+        following form::
+
+            'name': {'value': value_as_array,
+                     'offset': offset_in_bytes
+                     'description': description_string}
+
+        All ``'value'``'s are `numpy.ndarray`'s with at least one
+        dimension.
+
+    References
+    ----------
+    [Che+2015] Cheng, A et al. *MRC2014: Extensions to the MRC format header
+    for electron cryo-microscopy and tomography*. Journal of Structural
+    Biology, 129 (2015), pp 146--150.
+    """
+    # Positional args
+    num_cells = [int(n) for n in num_cells]
+    kind, kind_in = str(kind).lower(), kind
+    if kind not in ('volume', 'projections'):
+        raise ValueError("`kind '{}' not understood".format(kind_in))
+
+    # Keyword args
+    extent = kwargs.pop('extent', num_cells)
+    axis_order = kwargs.pop('axis_order', (0, 1, 2))
+    if tuple(axis_order) not in permutations((0, 1, 2)):
+        raise ValueError('`axis_order` must be a permutation of (0, 1, 2), '
+                         'got {}'.format(axis_order))
+    dmin = kwargs.pop('dmin', 1.0)
+    dmax = kwargs.pop('dmax', 0.0)
+    dmean = kwargs.pop('dmean', min(dmin, dmax) - 1.0)
+    rms = kwargs.pop('rms', -1.0)
+    mrc_version = kwargs.pop('mrc_version', (2014, 0))
+    if len(mrc_version) != 2:
+        raise ValueError('`mrc_version` must be a sequence of length 2, got '
+                         '{}'.format(mrc_version))
+
+    # Text labels: fill each label up with whitespace to 80 characters.
+    # Create the remaining labels as 80 * '\x00'
+    text_labels_in = kwargs.pop('text_labels', [])
+    nlabl = len(text_labels_in)
+    if nlabl > 10:
+        raise ValueError('expexted maximum of 10 labels, got {} labels'
+                         ''.format(nlabl))
+    text_labels = [str(label).ljust(80) for label in text_labels_in]
+    if any(len(label) > 80 for label in text_labels):
+        raise ValueError('labels cannot have more than 80 characters each')
+
+    # Convert to header-friendly form. Names are required to match
+    # exactly the header field names, and all of them must exist,
+    # so that `eval` below succeeds for all fields.
+    nx, ny, nz = [np.array(n, dtype='int32').reshape([1]) for n in num_cells]
+    mode = np.array(NPY_DTYPE_TO_MRC_MODE[np.dtype(dtype)],
+                    dtype='int32').reshape([1])
```

Comment:
Makes it a 1d array, not 0d. This is the easiest way of saying "I want a 1d array of exactly 1 element".